### ISSUE TITLE
2.8 loosing components after upgrade

### ISSUE THREAD

**Issue by rwidmark** [ironmansoftware/powershell-universal#933]
### Steps to Reproduce

Upgraded from 2.7.3 to 2.8 then you loose your components.

### Expected behavior

```console
That PSU will regonized the components from dashboards.ps1 file and include them.
```

### Actual behavior

```console
When upgrading from 2.7.3 I did have the components and everything was working, when I started 2.8 for the first time the components was not added to the dashboard but it was still present as it should in dashboards.ps1 file. I needed to go in to /admin and look at the dashboards and components to re-add them there.
After I did re-added them nothing was changed in the dashboards.ps1 file but now everything was working.

This is strange and I have around 15 dashboards and if I update our prod server then I have much work to do to re-add all of them manually so I'm hoping for a fix for this.
```

### Environment data

2.8, Windows server 2022

**Comment by adamdriscoll** [comment]
I was misunderstanding this issue but I can reproduce this. I'll get it resolved and have you verify it before I release a 2.8.1.

**Comment by adamdriscoll** [comment]
This issue has been mentioned on **Ironman Software Forums**. There might be relevant details there:

https://forums.ironmansoftware.com/t/psu-2-8-0-components-issue/6595/2

**Comment by adamdriscoll** [comment]
We have nightly builds available to resolve this issue. Please let me know if it works for you. 

https://imsreleases.blob.core.windows.net/universal-nightly/1812995419/PowerShellUniversal.2.8.1.msi
https://imsreleases.blob.core.windows.net/universal-nightly/1812995419/Universal.win7-x64.2.8.1.zip

**Comment by rwidmark** [comment]
> We have nightly builds available to resolve this issue. Please let me know if it works for you. 
> 
> 
> 
> https://imsreleases.blob.core.windows.net/universal-nightly/1812995419/PowerShellUniversal.2.8.1.msi
> 
> https://imsreleases.blob.core.windows.net/universal-nightly/1812995419/Universal.win7-x64.2.8.1.zip

I'll try it out tomorrow

**Comment by wsl2001** [comment]
you need to remove the components from the dashboard file, the new 2.8.1 version will automatically load all default components and if you have a custom one just add it to the pkg components under the universaldashboard folder and it will be loaded as well.

you can ignore the components button in PSU cause it will show none included but they are all loaded and running.

adam is going to fix the status in version 2.9.0 so it wont cause confusion.

**Comment by rwidmark** [comment]
> you need to remove the components from the dashboard file, the new 2.8.1 version will automatically load all default components and if you have a custom one just add it to the pkg components under the universaldashboard folder and it will be loaded as well.
> 
> 
> 
> you can ignore the components button in PSU cause it will show none included but they are all loaded and running.
> 
> 
> 
> adam is going to fix the status in version 2.9.0 so it wont cause confusion.

I know but that was not the original issue. The original issue was no components was loading.
I have not tried 2.8.1 yet I'll do that tonight. Had some other priorities today. VMware know how to implement bugs ;)

**Comment by rwidmark** [comment]
@adamdriscoll still having issues that the dashboard are not reloading when pushing a change.
I'm stuck in production with 2.7.3 until this is fixed :(
Still have the issue with the components also that when the dashboard do update from a git I need to restart the service to get the components back.

**Comment by adamdriscoll** [comment]
@rstolpe - Is that a different issue than the components not loading?

**Comment by rwidmark** [comment]
> @rstolpe - Is that a different issue than the components not loading? 

@adamdriscoll  can't say, whit 2.8.1 the dashboard don't reload when I push change with git. 9/10 times it don't reload and if it reloads I still need to restart the service to get the components to get added again.

Same issue with my work environment and my own test environment in azure.

I can try it more later tonight (3h until night here in Sweden) I have the kids now :)

**Comment by rwidmark** [comment]
So I have done some more testing and sometimes the components are correctly added after updating with git and sometimes it's not. Sometime some dashboards have added the components correctly after and some others have not.
Same thing with my work PSUs and my private once in Azure. Both are running win2022 servers. At work we have three servers with PSU same with all of them.

I'm not sure of this but this seem to only happen when you change in the components and not if you do changes in pages, but I'm not sure.

I did change dashboards.ps1 file and I did push it with GIT and the service did not restart so I needed to restart the PSU service manuell on the servers.

@adamdriscoll

**Comment by adamdriscoll** [comment]
For the components that aren't getting added, are they part of your repository that you are syncing with git?

I think I have identified an issue where we would actually read the components after reading the dashboard which could cause them to load. I've also identified a threading issue where, during a git sync, we could call the configuration service multiple times for the same changes and I'm not sure what affect that would have. I've fixed these problems for tonight's nightly build. 

I still need to do more testing on the git sync to see if I can reproduce this behavior. That said, this behavior was in 2.7 as well so I'm not sure if this is actually your problem.

**Comment by rwidmark** [comment]
> For the components that aren't getting added, are they part of your repository that you are syncing with git?
> 
> 
> 
> I think I have identified an issue where we would actually read the components after reading the dashboard which could cause them to load. I've also identified a threading issue where, during a git sync, we could call the configuration service multiple times for the same changes and I'm not sure what affect that would have. I've fixed these problems for tonight's nightly build. 
> 
> 
> 
> I still need to do more testing on the git sync to see if I can reproduce this behavior. That said, this behavior was in 2.7 as well so I'm not sure if this is actually your problem.  

Yes, the components are in the repo.

This issue started with 2.7.4.

I'll download the build tomorrow and see if it's solved.

**Comment by rwidmark** [comment]
@adamdriscoll this issue are also gone / fixed in PSU 2.9.0 nightly that I did download today.
I'll close the issue
